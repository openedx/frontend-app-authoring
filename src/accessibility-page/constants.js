export const COMMUNITY_ACCESSIBILITY_LINK = 'https://www.edx.org/accessibility';
export const ACCESSIBILITY_EMAIL = 'accessibility@edx.org';
