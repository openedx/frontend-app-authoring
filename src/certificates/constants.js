import { v4 as uuid } from 'uuid';

// eslint-disable-next-line import/prefer-default-export
export const defaultCertificate = {
  courseTitle: '',
  signatories: [{
    id: `local-${uuid()}`,
    name: '',
    title: '',
    organization: '',
    signatureImagePath: '',
  }],
};
