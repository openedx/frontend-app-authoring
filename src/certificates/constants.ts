import { v4 as uuid } from 'uuid';

export const defaultCertificate = {
  courseTitle: '',
  signatories: [{
    id: `local-${uuid()}`,
    name: '',
    title: '',
    organization: '',
    signatureImagePath: '',
  }],
};
