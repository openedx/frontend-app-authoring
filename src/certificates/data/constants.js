export const MODE_STATES = {
  noModes: 'no_modes',
  noCertificates: 'no_certificates',
  view: 'view',
  editAll: 'edit_all',
  create: 'create',
};

export const ACTIVATION_MESSAGES = {
  activating: 'Activating',
  deactivating: 'Deactivating',
};
