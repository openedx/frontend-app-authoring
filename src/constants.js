export const DATE_FORMAT = 'MM/dd/yyyy';
export const TIME_FORMAT = 'HH:mm';
export const DATE_TIME_FORMAT = 'YYYY-MM-DDTHH:mm:ss\\Z';
export const COMMA_SEPARATED_DATE_FORMAT = 'MMMM D, YYYY';
export const DEFAULT_EMPTY_WYSIWYG_VALUE = '<p>&nbsp;</p>';
export const STATEFUL_BUTTON_STATES = {
  default: 'default',
  pending: 'pending',
  error: 'error',
};

export const USER_ROLES = {
  admin: 'instructor',
  staff: 'staff',
};

export const BADGE_STATES = {
  danger: 'danger',
  secondary: 'secondary',
};

export const NOTIFICATION_MESSAGES = {
  saving: 'Saving',
  duplicating: 'Duplicating',
  deleting: 'Deleting',
};

export const DEFAULT_TIME_STAMP = '00:00';

export const COURSE_CREATOR_STATES = {
  unrequested: 'unrequested',
  pending: 'pending',
  granted: 'granted',
  denied: 'denied',
  disallowedForThisSite: 'disallowed_for_this_site',
};
