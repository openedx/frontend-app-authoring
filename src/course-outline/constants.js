export const CHECKLIST_FILTERS = {
  ALL: 'ALL',
  SELF_PACED: 'SELF_PACED',
  INSTRUCTOR_PACED: 'INSTRUCTOR_PACED',
};

export const LAUNCH_CHECKLIST = {
  data: [
    {
      id: 'welcomeMessage',
      pacingTypeFilter: CHECKLIST_FILTERS.ALL,
    },
    {
      id: 'gradingPolicy',
      pacingTypeFilter: CHECKLIST_FILTERS.ALL,
    },
    {
      id: 'certificate',
      pacingTypeFilter: CHECKLIST_FILTERS.ALL,
    },
    {
      id: 'courseDates',
      pacingTypeFilter: CHECKLIST_FILTERS.ALL,
    },
    {
      id: 'assignmentDeadlines',
      pacingTypeFilter: CHECKLIST_FILTERS.INSTRUCTOR_PACED,
    },
    {
      id: 'proctoringEmail',
      pacingTypeFilter: CHECKLIST_FILTERS.ALL,
    },
  ],
};

export const BEST_PRACTICES_CHECKLIST = {
  data: [
    {
      id: 'videoDuration',
      pacingTypeFilter: CHECKLIST_FILTERS.ALL,
    },
    {
      id: 'mobileFriendlyVideo',
      pacingTypeFilter: CHECKLIST_FILTERS.ALL,
    },
    {
      id: 'diverseSequences',
      pacingTypeFilter: CHECKLIST_FILTERS.ALL,
    },
    {
      id: 'weeklyHighlights',
      pacingTypeFilter: CHECKLIST_FILTERS.SELF_PACED,
    },
    {
      id: 'unitDepth',
      pacingTypeFilter: CHECKLIST_FILTERS.ALL,
    },
  ],
};
