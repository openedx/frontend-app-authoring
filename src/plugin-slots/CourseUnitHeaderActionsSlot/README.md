# CourseUnitHeaderActionsSlot

### Slot ID: `course_unit_header_actions_slot`
### Plugin Props:

* `headerNavigationsActions` - Object. See the prop-types definition in the component for details.
* `category` - String.
* `unitTitle` - String. Title of the Unit
* `isUnitVerticalType` - Boolean. Flag indicating if the Unit is VerticalBlock type. It's false for others like Library Blocks.
* `verticalBlocks` - Array. List of the vertical blocks.

## Description

The slot is positioned in the `SubHeader` section of the Course Unit editor page. It is suitable for adding action buttons.

By default, the slot contains the action buttons like **View Live Version** and **Preview**.

## Example

![Screenshot of an extra button in slot](./images/mybutton_added_to_slot.png)

The following example configuration inserts an extra button to the header as shown above.

```js
import { DIRECT_PLUGIN, PLUGIN_OPERATIONS } from '@openedx/frontend-plugin-framework';
import { Button } from '@openedx/paragon';

const MyButton = () => (
  <Button>üê£</Button>
);

const config = {
  pluginSlots: {
    course_unit_header_actions_slot: {
      keepDefault: true,
      plugins: [
        {
          op: PLUGIN_OPERATIONS.Insert,
          widget: {
            id: 'my-extra-button',
            priority: 60,
            type: DIRECT_PLUGIN,
            RenderWidget: MyButton,
          },
        },
      ]
    }
  },
}

export default config;
```
