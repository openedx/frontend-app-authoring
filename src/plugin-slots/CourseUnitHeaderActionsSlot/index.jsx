import React from 'react';
import PropTypes from 'prop-types';

import { PluginSlot } from '@openedx/frontend-plugin-framework';

import HeaderNavigations from 'CourseAuthoring/course-unit/header-navigations/HeaderNavigations';
import { COURSE_BLOCK_NAMES } from 'CourseAuthoring/constants';

const CourseUnitHeaderActionsSlot = ({
  headerNavigationsActions, category, unitTitle, verticalBlocks,
}) => {
  const isUnitVerticalType = category === COURSE_BLOCK_NAMES.vertical.id;
  return (
    <PluginSlot
      id="course_unit_header_actions_slot"
      pluginProps={{
        headerNavigationsActions, category, unitTitle, isUnitVerticalType, verticalBlocks,
      }}
    >
      <HeaderNavigations headerNavigationsActions={headerNavigationsActions} category={category} />
    </PluginSlot>
  );
};

CourseUnitHeaderActionsSlot.propTypes = {
  headerNavigationsActions: PropTypes.shape({
    handleViewLive: PropTypes.func.isRequired,
    handlePreview: PropTypes.func.isRequired,
    handleEdit: PropTypes.func.isRequired,
  }).isRequired,
  category: PropTypes.string.isRequired,
  unitTitle: PropTypes.string.isRequired,
  verticalBlocks: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string.isRequired,
      blockId: PropTypes.string.isRequired,
      name: PropTypes.string.isRequired,
      blockType: PropTypes.string.isRequired,
    }),
  ).isRequired,
};

export default CourseUnitHeaderActionsSlot;
