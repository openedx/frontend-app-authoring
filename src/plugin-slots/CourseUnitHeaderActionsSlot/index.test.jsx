import { shallow } from '@edx/react-unit-test-utils';
import renderer from 'react-test-renderer';

import { COURSE_BLOCK_NAMES } from 'CourseAuthoring/constants';
import CourseUnitHeaderActionsSlot from '.';

jest.mock('CourseAuthoring/course-unit/header-navigations/HeaderNavigations', () => 'HeaderNavigations');

jest.mock('@openedx/frontend-plugin-framework', () => ({
  PluginSlot: 'PluginSlot',
}));

// Slot properties that are required for the default HeaderNavigation component
const headerNavProps = {
  headerNavigationsActions: {
    handleViewLive: jest.fn(),
    handlePreview: jest.fn(),
    handleEdit: jest.fn(),
  },
  category: 'vertical',
  unitTitle: 'Mock Unit',
  verticalBlocks: [],
};

describe('CourseUnitHeaderActionsSlot', () => {
  beforeEach(() => jest.resetAllMocks());

  test('pluginProps are set correctly', () => {
    const wrapper = shallow(<CourseUnitHeaderActionsSlot unitTitle="Mock Title" verticalBlocks={[]} {...headerNavProps} />);
    expect(wrapper.snapshot).toMatchSnapshot();
  });

  test('isUnitVerticalType is set correctly', () => {
    let component = renderer.create(
      <CourseUnitHeaderActionsSlot
        unitTitle="Mock Title"
        verticalBlocks={[]}
        category="library"
        headerNavigationsActions={headerNavProps.headerNavigationsActions}
      />,
    );
    expect(component.toJSON().props.pluginProps.isUnitVerticalType).toEqual(false);

    component = renderer.create(
      <CourseUnitHeaderActionsSlot
        unitTitle="Mock Title"
        verticalBlocks={[]}
        category={COURSE_BLOCK_NAMES.vertical.id}
        headerNavigationsActions={headerNavProps.headerNavigationsActions}
      />,
    );
    expect(component.toJSON().props.pluginProps.isUnitVerticalType).toEqual(true);
  });
});
