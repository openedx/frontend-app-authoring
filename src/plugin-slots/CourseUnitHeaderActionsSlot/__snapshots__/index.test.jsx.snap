// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CourseUnitHeaderActionsSlot pluginProps are set correctly 1`] = `
<PluginSlot
  id="course_unit_header_actions_slot"
  pluginProps={
    {
      "category": "vertical",
      "headerNavigationsActions": {
        "handleEdit": [MockFunction],
        "handlePreview": [MockFunction],
        "handleViewLive": [MockFunction],
      },
      "isUnitVerticalType": true,
      "unitTitle": "Mock Unit",
      "verticalBlocks": [],
    }
  }
>
  <HeaderNavigations
    category="vertical"
    headerNavigationsActions={
      {
        "handleEdit": [MockFunction],
        "handlePreview": [MockFunction],
        "handleViewLive": [MockFunction],
      }
    }
  />
</PluginSlot>
`;
